typescript/Cute.ts

```diff
@@ -0,0 +1,53 @@
+module Cute {
+  export class Cutie {
+    private takeDecision(): boolean {
+      return parseInt((Math.random() * 100).toString()) % 2 === 0;
+    }
+
+    public tryToHug(): void {
+      try {
+        this.listen("Can I hug you?");
+        if (this.takeDecision()) {
+          this.say("For sure!");
+          this.hug();
+        } else {
+          throw new TypeError;
+        }
+      } catch (e) {
+        this.say("No, I don't like");
+        this.pleaseCanIHugYou();
+      }
+    }
+
+    public pleaseCanIHugYou(): void {
+      this.listen("Please, please, can I do it?");
+      if (this.takeDecision()) {
+        this.say("Ok, I allow it");
+        this.hug();
+      } else {
+        this.say("I SAID NO U PERVERT!");
```

Comment:
I'm a bit uncomfortable with the overall design of this one. In a model where we're seeking enthusiastic consent, it's better to avoid asking the same question multiple times with more pressure in hopes of getting a different answer. "Allowing" something isn't enough -- users should feel enthusiastic about the idea, and communicate their consent clearly and happily.